src/login/oidc/redirectHandler/AuthCodeRedirectHandler.ts

```diff
@@ -56,93 +49,38 @@ export default class AuthCodeRedirectHandler implements IRedirectHandler {
     }
     const url = new URL(redirectUrl, true);
     const localUserId = url.query.state as string;
-    const [
-      codeVerifier,
-      issuer,
-      clientId,
-      redirectUri,
-      clientSecret
-    ] = await Promise.all([
+    const [codeVerifier, redirectUri] = await Promise.all([
       (await this.storageUtility.getForUser(
         localUserId,
         "codeVerifier",
         true
       )) as string,
-      (await this.storageUtility.getForUser(
-        localUserId,
-        "issuer",
-        true
-      )) as string,
-      (await this.storageUtility.getForUser(
-        localUserId,
-        "clientId",
-        true
-      )) as string,
       (await this.storageUtility.getForUser(
         localUserId,
         "redirectUri",
         true
-      )) as string,
-      (await this.storageUtility.getForUser(
-        localUserId,
-        "clientSecret"
       )) as string
     ]);
 
-    const issuerConfig = (await this.issuerConfigFetcher.fetchConfig(
-      new URL(issuer)
-    )) as IIssuerConfig;
+    /* eslint-disable @typescript-eslint/camelcase */
+    await this.tokenRequester.request(localUserId, {
+      grant_type: "authorization_code",
+      code_verifier: codeVerifier as string,
+      code: url.query.code as string,
+      redirect_uri: redirectUri as string
+    });
+    /* eslint-enable @typescript-eslint/camelcase */
 
-    const tokenRequestInit: RequestInit = {
-      method: "POST",
-      headers: {
-        DPoP: await this.dpopHeaderCreator.createHeaderToken(
-          issuerConfig.tokenEndpoint,
-          "POST"
-        ),
-        "content-type": "application/x-www-form-urlencoded"
-      },
-      /* eslint-disable @typescript-eslint/camelcase */
-      body: formurlencoded({
-        client_id: clientId as string,
-        grant_type: "authorization_code",
-        code_verifier: codeVerifier as string,
-        code: url.query.code as string,
-        redirect_uri: redirectUri as string
-      })
-      /* eslint-enable @typescript-eslint/camelcase */
-    };
+    url.set("query", {});
 
-    if (clientSecret) {
-      (tokenRequestInit.headers as Record<
-        string,
-        string
-      >).Authorization = `Basic ${this.btoa(`${clientId}:${clientSecret}`)}`;
+    const session = await this.sessionCreator.getSession(localUserId);
+    if (!session) {
+      throw new Error("There was a problem creating a session.");
```

Comment:
Yikes! This seems no better than the dreaded "Server error" message! Is there any means to determine what the problem may have been (with the acknowledgement that less info is better from a security perspective, but I'd hope *some* indication of the problem might be helpful here)...?